### PR TITLE
Vulcan prowler improve output

### PR DESCRIPTION
**Pull request by manelmontilla**
This PR modifies the vulcan-prowler-check so it:
- Receives and option called "security-level" that determines which CIS Level, 1 or 2, the check uses to assets if an AWS is compliant with.
- Generates an "improved" version of the finding generated when the AWS account has any failed a or more CIS controls.

### CODE REVIEW

File: cmd/vulcan-prowler/main.go [hunk 1]
```diff
@@ -325,16 +363,31 @@ func fillCisLevelVuln(v *report.Vulnerability, r *prowlerReport, alias string, s
...
 		}
 	}
-
+	sort.Slice(rows, func(i, j int) bool {
+		return rows[i].score > rows[j].score
+	})
+	for _, r := range rows {
+		fcTable.Rows = append(fcTable.Rows, r.row)
+	}
 	v.Resources = append(CISCompliance.Resources, fcTable)
```
Comment (filewalkwithme) [inline cmd/vulcan-prowler/main.go:391]: I know that this line is not part of this PR but I want to suggest a change:

Why not use the following form:
```
v.Resources = append(v.Resources, fcTable)
```

instead of:
```
v.Resources = append(CISCompliance.Resources, fcTable)
```
?

I am saying that because Golang docs for the `Append` function says that we should try to use the same variable as source and destination:

```
It is therefore necessary to store the result of append, often in the variable holding the slice itself
```
https://golang.org/pkg/builtin/#append

Again, this is just a suggestion.

File: cmd/vulcan-prowler/main.go [hunk 2]
```diff
@@ -42,6 +43,8 @@ var (
 		"cislevel2",
 	}
 
+	// CISCompliance is the vulnerability generated by the check when the it
```
Comment (julianvilas) [inline cmd/vulcan-prowler/main.go:46]: `s/when the it/when it/`

File: cmd/vulcan-prowler/main.go [hunk 3]
```diff
@@ -67,9 +70,12 @@ var (
 			"https://github.com/toniblyx/prowler",
 			"https://www.cisecurity.org/benchmark/amazon_web_services/",
 		},
-		Score: report.SeverityThresholdMedium,
+		Score: report.SeverityThresholdLow,
 	}
 
+	// CISLevel1Compliance is the vulnerability generated by the check when the
```
Comment (julianvilas) [inline cmd/vulcan-prowler/main.go:76]: Same than above

File: cmd/vulcan-prowler/main.go [hunk 4]
```diff
@@ -97,9 +103,12 @@ var (
 			"https://github.com/toniblyx/prowler",
 			"https://www.cisecurity.org/benchmark/amazon_web_services/",
 		},
-		Score: report.SeverityThresholdLow,
+		Score: report.SeverityThresholdMedium,
 	}
 
+	// CISLevel2Compliance is the vulnerability generated by the check when the
```
Comment (julianvilas) [inline cmd/vulcan-prowler/main.go:109]: Same than above

File: cmd/vulcan-prowler/main.go [hunk 5]
```diff
@@ -127,9 +136,12 @@ var (
 			"https://github.com/toniblyx/prowler",
 			"https://www.cisecurity.org/benchmark/amazon_web_services/",
 		},
-		Score: report.SeverityThresholdMedium,
+		Score: report.SeverityThresholdLow,
```
Comment (julianvilas) [inline cmd/vulcan-prowler/main.go:139]: Why the severity of the issue for level 2 is lower than for level 1?

File: cmd/vulcan-prowler/main.go [hunk 6]
```diff
@@ -325,16 +363,31 @@ func fillCisLevelVuln(v *report.Vulnerability, r *prowlerReport, alias string, s
...
+			c := controlRow{row, cinfo.Severity}
+			rows = append(rows, c)
+			fallthrough
+		default:
+			total++
 		}
 	}
-
+	sort.Slice(rows, func(i, j int) bool {
+		return rows[i].score > rows[j].score
```
Comment (julianvilas) [inline cmd/vulcan-prowler/main.go:386]: Just a silly thing, but in this case the order won't be the same in different executions. I'd put more cases for the sort, for example sorting by the control number if the score is the same.

**Review by julianvilas — CHANGES_REQUESTED** [review]

File: cmd/vulcan-prowler/main.go (hunk 1 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:391]: I completely agree thanks for the suggestion!!

File: cmd/vulcan-prowler/main.go (hunk 2 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:46]: thanks fixed

File: cmd/vulcan-prowler/main.go (hunk 3 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:76]: thanks fixed

File: cmd/vulcan-prowler/main.go (hunk 4 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:109]: thanks fixed

File: cmd/vulcan-prowler/main.go (hunk 5 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:139]: That's a good question, we agreed on considering not being compliant with the CIS Level 2 not as bad as not being compliant with the CIS Level 1, which is more basic. It is true that there is no clear solution here.
What's your opinion regarding this?

File: cmd/vulcan-prowler/main.go (hunk 6 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:386]: That's a very good idea indeed thanks!!

File: cmd/vulcan-prowler/main.go [hunk 7]
```diff
@@ -360,6 +418,8 @@ func parseControl(raw string) (control string, description string, err error) {
 	// parts[0] = [check13 .
 	control = strings.Replace(parts[0], "[check", "", -1)
 	// control = 13 .
+	control = control[:1] + "." + control[1:]
```
Comment (filewalkwithme) [inline cmd/vulcan-prowler/main.go:421]: One last thing, if `control` is empty, this command will triggers a `slice bounds out of range` error. Can we put a protection/check for that?

Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:421]: I think it is a good suggestion,  the prowler tool must always return a non empty string, but it is true that it is better to check for that precondition.

**Review by filewalkwithme — APPROVED** [review]
Thanks @manelmontilla ! LGTM

**Review by julianvilas — APPROVED** [review]
